fsspec/implementations/chained.py

```diff
@@ -0,0 +1,9 @@
+from typing import ClassVar
+
+from fsspec import AbstractFileSystem
+
+__all__ = ("ChainedFileSystem",)
+
+
+class ChainedFileSystem(AbstractFileSystem):
```

Comment:
A comprehensive docstring here would be nice, and this should be added to the API docs